### PR TITLE
APPS/IO-DEMO: Replace completed field by checking UCS status

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Replace `completed` field by checking UCS status in UCP request.

## Why ?

Simplifies code and reduces the size of `ucx_request` structure.

## How ?

1. Remove `completed` field.
2. Set `UCS_INPROGRESS` to `status` field as initial value.
3. Check `status != UCS_INPRORGESS` instead of `completed`.

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: test/apps/iodemo/ucx_wrapper.cc [hunk 1]
```diff
@@ -1048,7 +1046,6 @@ void UcxConnection::common_request_callback(void *request, ucs_status_t status)
         UcxContext::request_release(r);
     } else {
         // not yet processed by "process_request"
-        r->completed = true;
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:1051]: remove 'else' part?

**Review by yosefe — COMMENTED** [review]
pls port to int3 as well

File: test/apps/iodemo/ucx_wrapper.cc (hunk 1 above)
Comment (dmitrygx) [reply test/apps/iodemo/ucx_wrapper.cc:1051]: done

**Comment by dmitrygx** [comment]
> pls port to int3 as well

@yosefe done - https://github.com/yosefe/ucx/pull/232

File: test/apps/iodemo/ucx_wrapper.cc [hunk 2]
```diff
@@ -292,7 +290,6 @@ void UcxContext::iomsg_recv_callback(void *request, ucs_status_t status,
                                      ucp_tag_recv_info *info)
 {
     ucx_request *r = reinterpret_cast<ucx_request*>(request);
-    r->completed   = true;
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:295]: shouldn't we set status here?
maybe it causes the test failures

File: test/apps/iodemo/ucx_wrapper.cc [hunk 3]
```diff
@@ -1038,17 +1035,15 @@ void UcxConnection::common_request_callback(void *request, ucs_status_t status)
 {
     ucx_request *r = reinterpret_cast<ucx_request*>(request);
 
-    assert(!r->completed);
+    assert(r->status == UCS_INPROGRESS);
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:1038]: i would write the code here like this:

```c
void UcxConnection::common_request_callback(void *request, ucs_status_t status)
{
    assert(status != UCS_INPROGRESS);

    ucx_request *r = reinterpret_cast<ucx_request*>(request);
    assert(r->status == UCS_INPROGRESS);

    r->status = status;
    if (r->callback) {
   ...
```

File: test/apps/iodemo/ucx_wrapper.cc [hunk 4]
```diff
@@ -290,6 +290,7 @@ void UcxContext::iomsg_recv_callback(void *request, ucs_status_t status,
                                      ucp_tag_recv_info *info)
 {
     ucx_request *r = reinterpret_cast<ucx_request*>(request);
+    r->status      = status;
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:293]: assert(status != UCS_INPROGRESS);

Comment (dmitrygx) [reply test/apps/iodemo/ucx_wrapper.cc:295]: done

**Review by yosefe — APPROVED** [review]

**Pull request merged**